### DISCUSSION TITLE
I have 2 account on same email, is it normal?

### DISCUSSION THREAD

**Discussion by NgoHuy** [community/community#69590] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have 2 account on github.com with same email: 1 managed user by my enterprise, 1 showed as my personal.
Is it normal on github.com? Because they dont relate to each other despite they use same email, also it cannot use same ssh key . I imagine that 2 accounts are on 2 tenants, but same git ssh account. Something is not true here.
Could you explain why it's allowed on github?

**Comment by moayad-star** [comment]
Yes, it's possible to have multiple GitHub accounts associated with the same email address. GitHub allows this because it understands that users may want to keep their personal and work-related repositories separate.

When you have two accounts with the same email, GitHub will still treat them as separate entities, and they won't automatically relate to each other. Each account will have its own repositories, settings, and SSH keys. This is designed to provide users with flexibility and the ability to manage different aspects of their work and personal projects separately.

If you want to use the same SSH key for both accounts, you can add the SSH key to each account individually. However, it's important to manage these accounts carefully to avoid any confusion or unintentional actions on the wrong account.

If you're experiencing specific issues or need further assistance with your GitHub accounts, you may want to reach out to GitHub support for personalized help.

**Reply by CristianHilohi-2** [reply]
Hi, I can't access one of the accounts because the password resets based on the email and then logs me in on the seconds account.

**Reply by CristianHilohi-2** [reply]
Oh, and git hub support is useless

**Reply by ghost** [reply]
same here @CristianHilohi-2

**Comment by NgoHuy** [comment]
@moayad-star thank  you for your answer, but I dont think it's really true. First, I cannot use same SSH key as authentication key, because we have only 1 user (git) when  we push the commit.
I know that web and git are different from system design. But when I login from my IdP, it shoud be mapped to github by some ways. The SAML's response contains user info, including email, and users are different - determined by emails. 
I think the email is only for personal login, and enterprise must login by username and email is just used for paring user+email in specific format email_org-name as username.

**Reply by moayad-star** [reply, marked as answer]
You're correct that the behavior of GitHub in an enterprise context can be more complex, and the handling of authentication and user mapping can vary depending on the configuration and integration with your Identity Provider (IdP).

In enterprise environments, GitHub often integrates with Single Sign-On (SSO) solutions, such as SAML-based authentication, to manage user access. The email address alone may not be the sole identifier, and the mapping between your IdP's user attributes and GitHub accounts can be customized.

It's possible that in your specific setup, GitHub is using a different method to map users between your enterprise and personal accounts, such as using a different username or a custom attribute from your IdP's SAML response.

To get a precise understanding of how your GitHub accounts are configured and mapped in your enterprise environment, I recommend reaching out to your organization's GitHub administrators or support team. They can provide insights into the specific setup and configuration that's in place for your accounts.

**Reply by NgoHuy** [reply]
Thank you very much, I just remember that email or UPN should not be used as identity from IdP, it should use resource id instead - IIRC.
Then this email could be reused somewhere in Github, and it's assigned as other single email.

**Reply by moayad-star** [reply]
If you encounter any problems regarding accessing or managing your account, please contact GitHub support to resolve any issues in the future
